tokio/src/io/util/fill_buf.rs

```diff
@@ -40,6 +40,12 @@ impl<'a, R: AsyncBufRead + ?Sized + Unpin> Future for FillBuf<'a, R> {
                 // Safety: This is necessary only due to a limitation in the
                 // borrow checker. Once Rust starts using the polonius borrow
                 // checker, this can be simplified.
+                //
+                // The safety of this transmute relies on the fact that the
+                // value of `reader` is `None` when we return in this branch,
+                // since otherwise the caller could poll us again after
+                // completion and access the mutable reference while the
+                // returned immutable reference still exists.
```

Comment:
nitpicky, not a big deal, but this is a super long sentence. it might be a bit more readable to split it:
```suggestion
                // value of `reader` is `None` when we return in this branch.
                // Otherwise the caller could poll us again after
                // completion, and access the mutable reference while the
                // returned immutable reference still exists.
```